scipy/stats/_axis_nan_policy.py

```diff
@@ -471,10 +461,8 @@ def hypotest_fun_out(*samples, **kwds):
                 # propagate nans in a sensible way
                 if any(contains_nans) and nan_policy == 'propagate':
                     res = np.full(n_outputs, np.nan)
```

Comment:
We don't call `result_unpacker` here because `np.full` already ensures that each output (a NaN) is an element of a sequence.

Reply:
> I'm not sure it's more confusing, but maybe it helps to twist things a bit: we _could_ call both

I thought about it and I think it would probably confuse someone reading this for first time.